### PR TITLE
Add test for DH initialization action

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Adds a test for the damped harmonic initalization action.

The only new commits are:
- Add test for DH initialization action

Depends on:
- [x] #2293 

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+  // Invoke the InitializeGhAnd3Plus1Variables action
+  ActionTesting::next_action<comp>(make_not_null(&runner), 0);
+  // Invoke the InitializeDampedHarmonic action
+  ActionTesting::next_action<comp>(make_not_null(&runner), 0);
+
+  tnsr::a<DataVector, Dim, Frame::Inertial> expected_gauge_h{};
+  tnsr::ab<DataVector, Dim, Frame::Inertial> expected_d4_gauge_h{};
+
+  GeneralizedHarmonic::gauges::damped_harmonic_rollon(
```
Comment (moxcodes) [inline tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp:145]: what a scary function call. (no clear change request I can think of)

**Review by moxcodes — APPROVED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please review

File: tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+template <size_t Dim, typename Metavariables>
+struct component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
```
Comment (geoffrey4444) [inline tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp:41]: Could this be a singleton instead? That would simplify the scary function call below somewhat

**Review by geoffrey4444 — APPROVED** [review]
Just one question, otherwise, looks reasonable to me

File: tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp (hunk 2 above)
Comment (nilsdeppe) [reply tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_InitializeDampedHarmonic.cpp:41]: We only have `MockArrayChare` in terms of the mocked types. The reason being partly that I never took the time to implement more and the other being that a singleton is just a 1-element array. Does that make sense?

**Review by kidder — APPROVED** [review]

**Pull request merged**